### PR TITLE
Added documentation for new course setup

### PR DESCRIPTION
**Pull request by Xiangmingchen**

### CODE REVIEW

**Comment by PradyumnaShome** [comment]
You also need to ensure all the repositories are named `NETID`

File: README.md [hunk 1]
```diff
@@ -15,3 +15,16 @@
...
+### New Course Setup
+- Ask on-demand managers to add your course to the on-demand database with at least one admin.
+  - Admins are able to modify roster and add assignments from the user interface.
+- Generate a token and add it to both broadway-on-demand and broadway's database for your course.
+- Create docker images for grading a single student
+  - On-demand will pass in the student's netid as `STUDENT_ID` and the timestamp as `DUE_DATE` to the all grader processes.
+- Add an assignment with the UI
+  - Assignment ID will be displayed and can not be changed once set.
+  - Max Runs along with Quota Type determine how many runs student get per day or total. Note that staff members always have one run for all assignments.
+  - Configuration is a json config provided to broadway. Refer to [broadway's Wiki](https://github.com/illinois-cs241/broadway/wiki/Configs). (The assignment config shouldn't have `STUDENT_ID` or `DUE_DATE` because those will be provided).
```
Comment (ayushr2) [inline README.md:28]: Can you also mention the due date format - due date is a string representation of the date so other course infra will need that format to decode the due date.

**Comment by ayushr2** [comment]
> You also need to ensure all the repositories are named NETID

Yeah we require the courses to have a GitHub Org of the name `{course-id}` and student repos should have the name `{student-id}`

File: README.md (hunk 1 above)
Comment (Xiangmingchen) [reply README.md:28]: What kind of format exactly? Do we need to add any time zone information to it?

Comment (ayushr2) [reply README.md:28]: https://github.com/illinois-cs241/broadway-on-demand/blob/1b75b1ea8473571e36398a5a8cdc860155750849/src/util.py#L114-L115

**Comment by Xiangmingchen** [comment]
> Yeah we require the courses to have a GitHub Org of the name {course-id} and student repos should have the name {student-id}

In the on-demand there is a separate field named `github-org` from id. Which course-id are we talking about here?

**Comment by PradyumnaShome** [comment]
> > Yeah we require the courses to have a GitHub Org of the name {course-id} and student repos should have the name {student-id}
> 
> In the on-demand there is a separate field named `github-org`. Which course-id are we talking about here?

Hmm I think that you guys are both talking about the same thing, then.

**Comment by Xiangmingchen** [comment]
> You also need to ensure all the repositories are named `NETID`

Is this used to grab the latest commit? In fact I thought that broadway would pass the SHA to the grader because every time the student click "Grade Now" they are looking at on-demand's commit, which could be a different one from what the AG actually grabs if the two logic are not the same....

**Comment by ayushr2** [comment]
> In the on-demand there is a separate field named github-org. Which course-id are we talking about here?

Yeah, interesting, it that used anywhere. From this line it looks like we use the course name to figure out the `github-org`.
https://github.com/illinois-cs241/broadway-on-demand/blob/master/src/ghe_api.py#L48

**Comment by ayushr2** [comment]
> Is this used to grab the latest commit? In fact I thought that broadway would pass the SHA to the grader because every time the student click "Grade Now" they are looking at on-demand's commit, which could be a different one from what the AG actually grabs if the two logic are not the same....

Broadway does not handle any of this GitHub related stuff. Broadway itself is not tied to GitHub. So Broadway does not figure out the commit sha. Broadway On-Demand is tied to GitHub (as you can see). So On-Demand itself tries to figure out the sha: https://github.com/illinois-cs241/broadway-on-demand/blob/1b75b1ea8473571e36398a5a8cdc860155750849/src/ghe_api.py#L41

**Comment by ayushr2** [comment]
Actually it looks like there is one more constraint. We also expect the course to set up a `_release` repo whose commits will be discarded while finding the latest commit sha...
https://github.com/illinois-cs241/broadway-on-demand/blob/1b75b1ea8473571e36398a5a8cdc860155750849/src/ghe_api.py#L13

**Comment by ayushr2** [comment]
> Actually it looks like there is one more constraint. We also expect the course to set up a _release repo whose commits will be discarded while finding the latest commit sha...

We should probably find a way to get rid of this constraint. Actually I have a PR open to fix this which is yet to be merged. #68

**Comment by Xiangmingchen** [comment]
> Broadway does not handle any of this GitHub related stuff. Broadway itself is not tied to GitHub.

But the containers on broadway calculates the latest SHA based on the `DUE_DATE` right?

> We should probably find a way to get rid of this constraint. Actually I have a PR open to fix this which is yet to be merged. #68

Agreed

**Comment by ayushr2** [comment]
> But the containers on broadway calculates the latest SHA based on the DUE_DATE right?

Yes

**Pull request merged**